triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -333,14 +345,19 @@ object Server {
         (dao, server)
       case Some(c) =>
         val dao = DbTriggerDao(c)
-        dao.readPackages match {
+        val server = new Server(ledgerConfig, restartConfig, secretKey, dao)
+        val recovery: Either[String, Unit] = for {
```

Comment:
I believe I brought this up before but let me bring it up again: I feel slightly uneasy about the fact that we are only reading from the DB on startup and then only write to it while using the in-memory state as the source of truth. That seems like it might be easy for things to get out of sync.
Did you consider abstracting over things such that if we run with postgres we always use postgres as the source of truth and don’t even keep the in-memory store?

Reply:
I'm not sure what you mean - the answer to your last question is we already do that. For tracking running triggers, the `RunningTriggerDao` abstracts over the read and write operations so we only use the in-memory store if we don't have a database and we only use the database if we have one. The one exception to this is the package map which we keep in memory in both cases because it's currently required to create a trigger Runner. The code here is specific to when the service is started up with a database provided. Does that address your concern?

Reply:
ah that sounds good, sorry for the confusion :+1: You might still be able to make startup a bit less specific to the postgres case.  If you run with an in-memory state reading the list of triggers will simply be empty on startup so there is no harm in going through the same logic.

Reply:
That's true - I started down that route since I already have `persistPackages` do nothing in the in-memory case. It seemed like maybe I would be forcing the abstraction continuing that way, but maybe it makes some sense. I'll consider for a future PR.